scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -373,7 +375,7 @@ object Defn {
       bounds: scala.meta.Type.Bounds,
       body: scala.meta.Type
   ) extends Defn with Member.Type {
-    checkFields(mods.exists(_.is[Mod.Opaque]))
+    // checkFields(mods.exists(_.is[Mod.Opaque]))
```

Comment:
So, since type match can have type bounds, I thinking of renaming the class to something like `TypeAliasWithBounds`

I was thinking of adding an optional field to Type, but it's impossible to say if we should add `bounds`, which are also used in the body of Type.